### ISSUE TITLE
[Combobox] Tab sequence gets stuck in popover interactive elements

### ISSUE THREAD

**Issue by indiesquidge** [reach/reach-ui#545]
### Current Behavior

When tabbing to interactive elements in the popover, the tab sequence gets stuck in the popover

<img width="75%" alt="combobox-no-tab-escape-from-popover" src="https://user-images.githubusercontent.com/3409645/79025017-74bb3680-7b39-11ea-8f0a-134bdcc4a88c.gif" />

Compare this to the popover itself, which does not track the tab sequence.

<img width="75%" alt="popover-tab-behavior" src="https://user-images.githubusercontent.com/3409645/79025058-94eaf580-7b39-11ea-9ada-89b2cc6d6ae7.gif" />

Interestingly, the tab sequence issue is not reproducible on the website examples, despite creating a clean slate with `cd website/ && rm -rf node_modules && yarn && yarn dev`.

<img width="75%" alt="combobox-website" src="https://user-images.githubusercontent.com/3409645/79025148-cf549280-7b39-11ea-87f8-a651de26c01f.gif" />

### Expected behavior

There is nothing mentioned on [WAI ARIA][1] about trapping focus in the listbox popover, so I would expect the tab sequence to predictably follow the browser default sequence, i.e. how the current Reach UI popover component behaves (as shown above).

### Reproducible example

[CodeSandbox Template](https://codesandbox.io/s/upbeat-meitner-me3n2)

Running the Storybook examples locally will also reproduce this issue

### Suggested solution(s)

Even after tinkering locally, its not clear to me why the tab sequence is trapped in the combobox popover (and also why it _isn't_ on the website examples).

### Your environment

| Software         | Name(s) | Version |
| ---------------- | ------- | ------- |
| Reach Package    | combobox | 0.10.0 |
| React            | - | 16.13.0 |
| Browser          | Chrome | 80.0.3987.163 |
| Assistive tech   | - | - |
| Node             | - | v13.12.0 |
| npm/yarn         | yarn | 1.22.4 |
| Operating System | macOS | 10.15.4 |

[1]: https://www.w3.org/TR/wai-aria-practices-1.2/#listbox-popup-keyboard-interaction

**Comment by indiesquidge** [comment]
This is perhaps a separate issue, but a fresh build of the website also warns that certain `@reach/utils` aren't exported (even though they are, and browsing my local `node_modules` even shows that they are).

![Screen Shot 2020-04-10 at 2 36 35 PM](https://user-images.githubusercontent.com/3409645/79028911-f2397380-7b46-11ea-84c0-ce27f8b1d128.png)

**Comment by chancestrickland** [comment]
There has been some discussion on how we handle this by default in various components. All of the ARIA examples I've seen that implement popovers with interactive content do in fact lock focus in the popover, though some UI specs don't mention it explicitly (particularly if the example doesn't use nested tabbable elements). My general feeling is aligned with yours here, and that's the default approach taken in `listbox` currently. I would like to get some more feedback from users in the a11y community here.

That said, no idea why the behavior wouldn't be consistent between the docs and local build. Haven't had a problem building the site locally on my end or in CI 🤔

**Comment by raunofreiberg** [comment]
I think the root of the problem lies in Popover. Hear me out!

All of the examples where tab sequence is not disrupted have focusable elements before and after the component. 

I added a sibling `input` element to the reproducible example and the tab sequence is correct: https://codesandbox.io/s/new-darkness-uipul

![gif](https://gyazo.com/9f233c35dc8d8f03c3bf5e2021c18524.gif)

<hr />

Alas, if you reduce the Popover example down to a single `textarea`, the same issue appears:
https://codesandbox.io/s/new-darkness-uipul?file=/src/index.js

I'll try to take a closer look later, but on first guess, the devil is somewhere here:

https://github.com/reach/reach-ui/blob/52d721cd643d1a6d24c7253ab24e114be262d092/packages/popover/src/index.tsx#L193

**Comment by raunofreiberg** [comment]
I dunno if this makes sense, but these changes to [`focusLastTabbableInPopover`](https://github.com/reach/reach-ui/blob/52d721cd643d1a6d24c7253ab24e114be262d092/packages/popover/src/index.tsx#L291) seem to fix tabbing back and forth between the trigger and elements inside the Popover.

```diff
function focusLastTabbableInPopover(event: KeyboardEvent) {
  const elements = popoverRef.current && tabbable(popoverRef.current);

+  // The actual last tabbable should be the trigger.
+  // We push it manually to the array because it is outside of the portal of popover.
+  triggerRef.current && elements?.push(triggerRef.current);

  const last = elements && elements[elements.length - 1];

  if (last) {
    event.preventDefault();
    last.focus();
  }
}
```

![gif](https://gyazo.com/b5ac7db6037e9e08bdb39b8a8f3491fa.gif)

**Comment by michaeldfoley** [comment]
I too believe that the issue relates to Popover, specifically cases where the `targetRef` is the last tabbable element outside the popover. When this is the case, `getElementAfterTrigger` returns the first tabbable element inside the portal causing the focus to stay trapped inside the popover. What we want is for `getElementAfterTrigger` to return an element only if it is not inside the popover.

I submitted a PR (#609) that should address the issue. I recreated the [original example](https://codesandbox.io/s/headless-brook-0zfd8) referenced in this issue and the focus now moves outside the popover as expected.
